cmd/nerdctl/run_network.go

```diff
@@ -164,12 +167,104 @@ func generateNetOpts(cmd *cobra.Command, dataStore, stateDir, ns, id string) ([]
 				ports = append(ports, pm...)
 			}
 		}
+	case nettype.Container:
+		if cmd.Flags().Changed("publish") {
+			return nil, nil, "", nil, fmt.Errorf("conflicting options: port publishing and the container type network mode")
+		}
+		if runtime.GOOS != "linux" {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can only works on linux")
+		}
+		if rootlessutil.IsRootless() {
+			return nil, nil, "", nil, fmt.Errorf("currently '--network=container:<container>' can't run rootless")
+		}
+		if len(netSlice) > 1 {
+			return nil, nil, "", nil, fmt.Errorf("only one network allowed using '--network=container:<container>'")
+		}
+		network := strings.Split(netSlice[0], ":")
+		if len(network) != 2 {
+			return nil, nil, "", nil, fmt.Errorf("invalid network: %s, should be \"container:<id|name>\"", netSlice[0])
+		}
+		containerName := network[1]
+		client, ctx, cancel, err := newClient(cmd)
+		if err != nil {
+			return nil, nil, "", nil, err
+		}
+		defer cancel()
+		var containerID string
+		walker := &containerwalker.ContainerWalker{
+			Client: client,
+			OnFound: func(ctx context.Context, found containerwalker.Found) error {
+				if found.MatchCount > 1 {
+					return fmt.Errorf("multiple containers found with prefix: %s", containerName)
+				}
+				containerID = found.Container.ID()
+				// share same hostname with the target container
+				cmd.Flags().Set("hostname", containerID[0:12])
```

Comment:
```
# docker run --rm -it --domainname baz.cyphar.com --hostname test --net container:nginx alpine sh
docker: Error response from daemon: conflicting options: hostname and the network mode.
```
conflicting options: hostname and the network mode.`

Reply:
will return an error for this situation.  `--hostname` and `--network=container:***` conflict

Reply:
Why need it yet?

Reply:
Not quite sure what you are referring to, the code here is for containers using the same hostname. docker did the same.

```
# docker run -d --name ng nginx:1.19-alpine
# docker run -d -it --name join  --network=container:ng alpine:3.13 sleep infinity
# docker exec ng cat /etc/hostname
7968dccea5f1
# docker exec join cat /etc/hostname
7968dccea5f1

```

Reply:
> will use the target container's hostname instead of Its ID.

Incorrect, they should share the same etchosts file.

Reply:
my experiment above shows they shared `/etc/hostname` too in docker, and `/etc/hosts` is already using the same file by `withCustomHosts(etcHostsPath)`

Reply:
> Why need it yet?

I don't understand why we need it

Reply:
> my experiment above shows they shared /etc/hostname too in docker, and /etc/hosts is already using(mount) the same file by withCustomHosts(etcHostsPath) in walker 😄

Sorry, using `buildResolvConf` and `AllocHostsFile` confuse me, we just need to share files, we don't need to create they.